sycl/test/check_device_code/matrix/matrix-nvptx-bfloat16-test.cpp

```diff
@@ -0,0 +1,208 @@
+// REQUIRES: cuda
+
+// RUN: %clangxx -fsycl-device-only -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3 -S -Xclang -emit-llvm %s -o -| FileCheck %s
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+using sycl::ext::oneapi::experimental::bfloat16;
```

Comment:
Is this compiling for you?
In my case, I have to explicitly include 
#include <sycl/ext/oneapi/experimental/bfloat16.hpp>
for it to recognize bfloat16.
The reason for that is that sycl.hpp does not have that include

Reply:
Yeah I thought that we weren't meant to add these experimental headers to sycl.hpp. Are you using this invocation:
```
// RUN: %clangxx -fsycl-device-only -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3 -S -Xclang -emit-llvm %s -o -| FileCheck %s
```

Reply:
It isn't meant to be run by the runtime.

Reply:
I did not try this code. But for other codes that use bfloat16, when I compile using dpcpp test.cpp, I get:
use of undeclared identifier 'bfloat16'.
I had to include bfloat16 header file.
Also, if you look at bfloat16 examples, they also include it: https://github.com/intel/llvm-test-suite/blob/intel/SYCL/BFloat16/bfloat16_type.hpp
If this include is required, we should add it to sycl.hpp but was wondering how come you did not need it for your test.

Reply:
I see. These tests all use the `SYCL_EXT_ONEAPI_MATRIX=3`macro that includes matrix-tensorcore.hpp which includes bfloat16.hpp. I think this is why the explicit bfloat16.hpp include isn't needed.

Reply:
Okay, this is why.
But we should be consistent WRT including this header file.
Shouldn't we add that include to sycl.hpp like the other features and remove the extra scattered includes from the other files?

Reply:
Happy to include it to sycl.hpp if it is decided that is what should happen for default extensions: However there is an argument against that because it is allowed (and there is) C++17 standard in the extension headers so long as these headers aren't included in sycl.hpp (c++17 not allowed in sycl.hpp for some reason).

So my understanding is we either 
a) Allow c++17 in extensions and make it responsibility of the user to include headers.
b) include default extensions in sycl.hpp but remove all c++17 usage.

Whatever is decided I think it should be enforced consistently.